gnd/src/main/java/com/google/android/gnd/repository/local/IntEnum.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.repository.local;
+
+import androidx.annotation.NonNull;
+import java8.util.stream.Stream;
+
+/**
+ * Common interface for Java enums with explicitly defined int representations. This is used instead
+ * of relying on enum ordinal values to prevent accidentally breaking backwards compatibility when
+ * adding and/or removing new enum values in the future.
+ */
+public interface IntEnum {
+  int intValue();
+
+  static <E extends IntEnum> int fromIntEnum(E enumValue, @NonNull E defaultValue) {
+    return enumValue == null ? defaultValue.intValue() : enumValue.intValue() ;
+  }
+
+  @NonNull
+  static <E extends Enum<E> & IntEnum> E toIntEnum(
+      @NonNull Stream<E> values, int intValue, @NonNull E defaultValue) {
```

Comment:
What's the use case for this? In particular, why does it take a `Stream` as an argument instead of, say, `Iterable`? That seems unusual to me (and depending on the use case, there might be a more idiomatic way to express this). Do you have a simple code snippet to show a typical usage pattern?

Reply:
TypeConverters were using this like so:
```java
  @NonNull
  @TypeConverter
  public static EntityState fromInt(int intValue) {
    return IntEnum.fromInt(stream(values()), intValue, UNKNOWN);
  }
```

Where `values()` is the array of const values defined for the `Enum`. This allows us to iterate (or stream, rather) over the array without the intermediate step of converting to an `Iterable`.  Unfortunately polymorphism doesn't work on Arrays in Java, so we can't pass `IntEnum[]` as a generic argument either.

Reply:
I would suggest just accepting an `Iterable` instead of a `Stream` (and convert to a stream to perform the operation internally). It doesn't feel like that much different at the call site to call `Arrays.asList(a)` instead of `Arrays.stream(a)`, and eliminates the possibility for weird behavior when a stream is passed across different method calls (e.g. if the stream is accessed concurrently and closed somewhere else while this method is still trying to read it). I think it's easier to understand how iterable behave in those situations than streams.

If operating on streams instead of iterables, this would be more idiomatic as a `Collector` -- I think this is roughly the correct signature:
```java
static <E> Collector<E, ?, E> matchingInt(int intValue, E defaultValue);
``` 
and then the caller can do something like
```java
return stream(values()).collect((IntEnum.matchingInt(intValue, UNKNOWN));
```

But implementing a `Collector` seems like more complexity than is really needed here.

Reply:
Good point about exposing streams directly. Even though this package private we should establishing precedent for potentially dangerous patterns. I actually could use generics here to pass the array, which seems to be the simplest option. Doing that instead.

Reply:
Did you make a change here?